sdks/java/core/src/main/java/org/apache/beam/sdk/coders/StringDelegateCoder.java

```diff
@@ -17,9 +17,14 @@
  */
 package org.apache.beam.sdk.coders;
 
+import org.apache.beam.sdk.coders.DelegateCoder.CodingFunction;
```

Comment:
I'm +1 on removing it, but -1 to inlining it. The original motivation was that users shouldn't have to write coders for very "smart strings" like `URI`, and it extends to any ubiquitous type.

A straightforward alternative to `StringDelegateCoder` (and actually my first approach, rejected via the argument "but we can't have a coder for everything") is to simply provide coders for important/ubiquitous types. Extensions and IO libraries should own their coders (the mechanisms for this today are not perfect, but that is a separate issue) but the SDK needs to own the heavy hitters. We are somewhat inconsistent in this area.

In a multi-language world, only the second approach is portable, so maybe it tips the scales. And it narrows the number of such types to consider: only those that have a meaningful cross-language definition. We may have been too hasty with the bignum coders to base them on the convenient accessors provided Java, when they should have binary specs that are sensible across languages.

But I maintain that the best number of lines I should have to write to be able to use a `PCollection<URI>` is zero and the maximum is one.